format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -70,45 +73,40 @@ protected AbstractFileSource(T config) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    config.validate();
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(collector);
 
-    Schema schema = config.getSchema();
-    FileFormat fileFormat = config.getFormat();
+    FileFormat fileFormat = getFileFormat();
+    Schema schema = null;
     if (fileFormat != null) {
```

Comment:
this shouldn't have been here either, format is required.

Reply:
format is macro enabled tho